### PR TITLE
Add deprecation warning for Spree.t

### LINKED ISSUE #2773: Deprecate Spree.t
So I'm currently working on a Solidus project and I saw that @jhawthorn has changed `Spree.t` => `t('spree.')`. I'm thinking we should deprecate `Spree.t` as I've been going into the project I've been on and changing `t('spree.')` to `Spree.t`; I'm regretting that now. I think a warning should begin to be emitted to programmers using `Spree.t` and remove it in Solidus 3 or 4.

Issue #2309

What are some thoughts on this?

### PR DESCRIPTION
**Pull request by JDutil**
fixes #2773

### CODE REVIEW

**Comment by kennyadsl** [comment]
@JDutil specs failure are actually related, can you please take a look? 🙏

File: core/spec/lib/i18n_spec.rb [hunk 1]
```diff
@@ -29,25 +29,35 @@
     I18n.reload!
   end
 
-  it "translates within the spree scope" do
-    expect(Spree.t(:foo)).to eql("bar")
-    expect(Spree.translate(:foo)).to eql("bar")
-  end
+  describe '#translate' do
+    after do
+      Spree::Deprecation.behavior = :raise
```
Comment (kennyadsl) [inline core/spec/lib/i18n_spec.rb]: Another approach could be stubbing the deprecation warning 

```
expect(Spree::Deprecation).to receive(:warn)
```

Comment (tvdeyen) [reply core/spec/lib/i18n_spec.rb]: Leaving it like that means that from the moment we ran this spec we always raise for deprecations even if we change this in the spec helper in the future. Either wrap the spec in a

```rb
around do |example|
  Spree::Deprecation.silence { example.run }
end
```

or do what @kennyadsl proposes.

Third possibility is to remove specs for deprecated methods.

**Comment by JDutil** [comment]
I prefer @kennyadsl solution will update :+1:

**Comment by kennyadsl** [comment]
@tvdeyen in this case I'd leave the specs since we are not really removing this functionality but just printing a deprecation warning. I think it's better to be sure stores and extensions that still relies on that method can work.

**Pull request merged**